repos/system_upgrade/el7toel8/actors/selinux/selinuxcontentscanner/libraries/library.py

```diff
@@ -0,0 +1,200 @@
+        os.mkdir(WORKING_DIRECTORY)
+        os.chdir(WORKING_DIRECTORY)
+    except OSError:
+        api.current_logger().info("Failed to access working directory! Aborting.")
+        return ([], [], [])
+
+    for (name, priority) in modules:
+        if priority == "200":
+            # Module on priority 200 was installed by an RPM
+            # Request $name-selinux to be installed on RHEL8
+            retain_rpms.append(name + "-selinux")
+            continue
+        if priority == "100":
+            # module from selinux-policy-* package - skipping
+            continue
+        # extract custom module and save it to SELinuxModule object
+        try:
+            run(['semodule', '-c', '-X', priority, '-E', name])
+            # check if the module contains invalid types and remove them if so
+            removed = checkModule(name + ".cil")
+
+            # get content of the module
+            try:
+                with open(name + ".cil", 'r') as cil_file:
+                    module_content = cil_file.read()
+            except OSError as e:
+                api.current_logger().info("Error reading %s.cil : %s", name, e.get("stderr", ""))
+                continue
+
+            semodule_list.append(SELinuxModule(
+                name=name,
+                priority=int(priority),
+                content=module_content,
+                removed=removed
+                )
+            )
+        except CalledProcessError:
+            api.current_logger().info("Module %s could not be extracted!", name)
+            continue
+        # rename the cil module file so that it does not clash
+        # with the same module on different priority
+        try:
+            os.rename(name + ".cil", "%s_%s" % (name, priority))
+        except OSError:
+            # TODO leapp.libraries.stdlib.api.current_logger()
+            # and move the method to a library
+            api.current_logger().info("Failed to rename module file.")
+    # this is necessary for check if container-selinux needs to be installed
+    try:
+        run(['semanage', 'export', '-f', 'semanage'])
+    except CalledProcessError:
+        pass
+    # Check if modules contain any type, attribute, or boolean contained in container-selinux and install it if so
+    # This is necessary since container policy module is part of selinux-policy-targeted in RHEL 7 (but not in RHEL 8)
+    try:
+        run(['grep', '-w', '-r', '-E', "|".join(CONTAINER_TYPES)], split=False)
+        # Request "container-selinux" to be installed since container types where used in local customizations
+        # and container-selinux policy was removed from selinux-policy-* packages
+        install_rpms.append("container-selinux")
+    except CalledProcessError:
+        # expected, ignore exception
+        pass
+
+    try:
+        os.chdir(wd)
+    except OSError:
+        pass
+    rmtree(WORKING_DIRECTORY, ignore_errors=True)
+
+    return (semodule_list, retain_rpms, install_rpms)
```

Comment:
I prefer to ensure that retain_rpms and install_rpms do not contain duplicit values regarding the implementation.

```suggestion
    return (semodule_list, list(set(retain_rpms)), list(set(install_rpms)))
```